ord_schema/interface/query.py

```diff
@@ -200,148 +390,27 @@ def __init__(self, dbname, user, password, host, port):
-            logging.info(command.as_string(cursor))
-            do_chiral_sss = 'true' if use_stereochemistry else 'false'
-            # NOTE(kearnes): We call rollback() to reset this change before
-            # exiting the context manager (which triggers a commit).
-            cursor.execute(sql.SQL('SET rdkit.do_chiral_sss=%s'),
-                           [do_chiral_sss])
-            cursor.execute(command, args)
-            reactions = []
-            for result in cursor:
-                _, serialized = result
-                reaction = reaction_pb2.Reaction.FromString(
-                    binascii.unhexlify(serialized.tobytes()))
-                reactions.append(reaction)
-            self._connection.rollback()
-        return dataset_pb2.Dataset(reactions=reactions)
-
-    def similarity_search(self, smiles, table, limit=100, threshold=0.5):
-        """Performs a Tanimoto similarity search.
-
-        Args:
-            smiles: Text SMILES.
-            table: Text SQL table name.
-            limit: Integer maximum number of matches to return.
-            threshold: Float similarity threshold.
-
-        Returns:
-            Dataset proto containing matched Reactions.
-        """
-        if 'reactions' in table:
-            column = 'rdfp'
-            function = 'reaction_difference_fp'
-        else:
-            column = 'mfp2'
-            function = 'morganbv_fp'
-        components = [
-            sql.SQL("""
-                SELECT DISTINCT A.reaction_id, A.serialized 
-                FROM reactions A 
-                INNER JOIN {} B ON A.reaction_id = B.reaction_id
-                WHERE B.{}%%{}(%s)""").format(sql.Identifier(*table.split('.')),
-                                              sql.Identifier(column),
-                                              sql.Identifier(function))
-        ]
-        args = [smiles]
-        if limit:
-            components.append(sql.SQL(' LIMIT %s'))
-            args.append(limit)
-        command = sql.Composed(components).join('')
-        with self._connection, self.cursor() as cursor:
-            logging.info(command.as_string(cursor))
-            # NOTE(kearnes): We call rollback() to reset this change before
-            # exiting the context manager (which triggers a commit).
-            cursor.execute(sql.SQL('SET rdkit.tanimoto_threshold=%s'),
-                           [threshold])
-            cursor.execute(command, args)
-            reactions = []
-            for result in cursor:
-                _, serialized = result
-                reaction = reaction_pb2.Reaction.FromString(
-                    binascii.unhexlify(serialized.tobytes()))
-                reactions.append(reaction)
-            self._connection.rollback()
-        return dataset_pb2.Dataset(reactions=reactions)
+        if return_ids:
+            return dataset_pb2.Dataset(reaction_ids=reactions.keys())
+        unserialized = []
+        for serialized in reactions.values():
```

Comment:
Made a dictionary then threw away the keys?

Could you skip a step and preserve the DB result order?

Reply:
The dictionary keys are used if `return_ids=True`. I don't think the result order is important.